google/cloud/bigquery/v2/minimal/internal/job_rest_stub.cc

```diff
@@ -64,11 +64,14 @@ StatusOr<InsertJobResponse> DefaultBigQueryJobRestStub::InsertJob(
   nlohmann::json json_payload;
   to_json(json_payload, request.job());
 
+  auto filtered_json = RemoveJsonKeysAndEmptyFields(json_payload.dump(),
+                                                    request.json_filter_keys());
```

Comment:
for the record, I don't think we ever `request.set_json_filter_keys(...)` so we are not removing any keys in this PR.